xml_converter/generators/cpp_templates/class_template.cpp

```diff
@@ -46,7 +46,7 @@ bool {{cpp_class}}::init_xml_attribute(rapidxml::xml_attribute<>* attribute, vec
     }
```

Comment:
Something seemed strange about having one branch of this `if` `elif` `else` use hardcoded `attribute, errors` while the others used `",".join(attribute_variable.args`. And it made me realize that this entire section is kinda confusingly laid out. Because of that there is a possible bug. Which is if the first attribute has a side effect, what happens? Or similarly "what happens if the first attribute is a compound value".


All three of these things check for `if (attributename == "{{value}}")` and then do other things based off of other checks.

This task has gotten so large at this point that it is probably not worth fixing in this task but the very next one should probably look at it. A later branch would also be able to take advantage of the jinja indentation pr to help make this more readable. Maybe something more like:


```cpp
    {% if i != 0 or n != 0 %}else {% endif %}if (attributename == "{{value}}" {
        {% if attribute_variable.attribute_type == "CompoundValue" and attribute_variable.compound_name != None %}
            this->{{attribute_variable.compound_name}}.{{attribute_variable.attribute_name}} = parse_float(attribute, errors);
            this->{{attribute_variable.compound_name}}_is_set = true;
        {% else %}
            this->{{attribute_variable.attribute_name}} = parse_{{attribute_variable.class_name}}({{", ".join(attribute_variable.args)}});
            this->{{attribute_variable.attribute_name}}_is_set = true;
        {% endif %}
        {% for side_effect in attribute_variable.side_effects %}
            this->{{side_effect}} = this->{{attribute_variable.class_name}}.side_effect_{{side_effect}};
            this->{{side_effect}}_is_set = true;
        {% endfor %}
    }
```

and if you dont like that inlined "else " then you can break it out to two lines

```cpp
    {% if i == 0 and n == 0 %}
        if (attributename == "{{value}}" {
    {% else %}
        else if (attributename == "{{value}}" {
    {% endif %}
        {% if attribute_variable.attribute_type == "CompoundValue" and attribute_variable.compound_name != None %}
            this->{{attribute_variable.compound_name}}.{{attribute_variable.attribute_name}} = parse_float(attribute, errors);
            this->{{attribute_variable.compound_name}}_is_set = true;
        {% else %}
            this->{{attribute_variable.attribute_name}} = parse_{{attribute_variable.class_name}}({{", ".join(attribute_variable.args)}});
            this->{{attribute_variable.attribute_name}}_is_set = true;
        {% endif %}
        {% for side_effect in attribute_variable.side_effects %}
            this->{{side_effect}} = this->{{attribute_variable.class_name}}.side_effect_{{side_effect}};
            this->{{side_effect}}_is_set = true;
        {% endfor %}
    }
```

Which is arguably more or arguably less readable.